File: Sorting/binary_search.js
const binarySearch = (arr,val)=>{
	if(arr.length<1) return false;
	if(arr.length === 1){
		if(val === arr[0]) return true;
		else return false;
	} 
	const mid = Math.floor(arr.length/2);
	const midVal = arr[mid];
	return (midVal === val)? true:(val < midVal)?binarySearch(arr.slice(0,mid),val): binarySearch(arr.slice(mid),val);
}

File: Sorting/BinarySearch.java
class BinarySearch {
  private static int binarySearch(int[] arr, int find, int l, int r){
    if(r<l) return -1;
    int mid = l+(r-l)/2;
    if(arr[mid] == find) return find;
    else if(arr[mid] > find) return binarySearch(arr, find, l, mid-1);
    else return binarySearch(arr, find, mid+1, r);
  }

  
  public static void main(String[] args) {
    int[] arr = new int[]{1,2,3,4,5,6};
    int find = 3;
    int result = binarySearch(arr, find, 0, arr.length-1);
    System.out.println("result = " + result);
  }
}